main/src/modules/contextualLayer/goalManager/src/goalManager.cpp

```diff
@@ -0,0 +1,216 @@
+#include "goalManager.h"
+#include "wrdac/subsystems/subSystem_recog.h"
+
+bool GoalManager::configure(yarp::os::ResourceFinder &rf)
+{
+    string moduleName = rf.check("name", Value("goalManager")).asString().c_str();
+    setName(moduleName.c_str());
+
+    yInfo() << moduleName << " : finding configuration files...";
+    period = rf.check("period", Value(0.1)).asDouble();
+
+    //Create an iCub Client and check that all dependencies are here before starting
+    bool isRFVerbose = false;
```

Comment:
If no `ICubClient` is needed, please remove these comments and the corresponding variable from the header file.